src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/activity/ActivityPageFactory.java

```diff
@@ -1,29 +1,24 @@
 package com.wikia.webdriver.pageobjectsfactory.componentobject.activity;
 
-import com.wikia.webdriver.pageobjectsfactory.componentobject.activity.creators.ActivityPageCreator;
-import org.openqa.selenium.WebDriver;
+
+import lombok.Getter;
 import org.openqa.selenium.WebElement;
-import javax.swing.text.html.HTML;
-import java.util.List;
 
+import java.util.ArrayList;
+import java.util.List;
 
 public class ActivityPageFactory {
-  private List<ActivityPageCreator> creators;
-  private WebElement activity;
-  private WebDriver driver;
-
-  public ActivityPageFactory(WebDriver driver, WebElement activity, List<ActivityPageCreator> creators) {
-    this.creators = creators;
-    this.activity = activity;
-    this.driver = driver;
+
+  @Getter
+  private List<Activity> activities = new ArrayList<>();
+
+  public ActivityPageFactory(List<WebElement> activities) {
+    this.activities = buildActivities(activities);
   }
 
-  public Activity createActivityPage() {
-    String activityClass = activity.getAttribute(HTML.Attribute.CLASS.toString());
-    if (!creators.stream().filter(c -> c.match(activityClass)).findAny().isPresent()) {
-      throw new IllegalArgumentException("Type of activity was not recognized");
-    }
-    ActivityPageCreator creator = creators.stream().filter(c -> c.match(activityClass)).findAny().get();
-    return creator.createPage(driver, activity);
+  private List<Activity> buildActivities(List<WebElement> activities) {
+    activities.forEach(element -> this.activities.add(new Activity(element)));
+    return this.activities;
```

Comment:
So this method works on a field, returns the field and the result of this method is assigned to the field.
Wouldn't this be equivalent?
```
return activities.map(Activity::new).collect(Collectors.toList());
```